server/services/v1/query_runner.go

```diff
@@ -546,14 +547,30 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
 		}
 	}
 
+	facets, err := qsearch.UnmarshalFacet(runner.req.Facet)
+	if err != nil {
+		return nil, ctx, err
+	}
+
+	pageSize := int(runner.req.PageSize)
+	if pageSize == 0 {
+		pageSize = defaultPerPage
+	}
+
 	searchQ := qsearch.NewBuilder().
 		Query(runner.req.Q).
 		SearchFields(searchFields).
-		PageSize(defaultPerPage).
+		Facets(facets).
+		PageSize(pageSize).
 		Filter(wrappedF).
 		Build()
 
-	rowReader, err := NewSearchReader(ctx, runner.searchStore, collection, searchQ)
+	var rowReader *SearchRowReader
+	if runner.req.Page != 0 {
```

Comment:
So first page index is 1?

Reply:
it is 1. I was thinking to make it 0, the only thing is clients need to make sure to use a pointer otherwise default value would be 0.